hw/syn/tools/dc/run-syn.tcl

```diff
@@ -28,29 +32,12 @@ define_design_lib WORK -path $WORKLIB
 
 # just compile the "core" toplevel at the moment
 # might want to switch to top_earlgrey_asic later on (with pads)
-set TOP_ENTITY [getenv "TOP_ENTITY"]
-
-# read source list generated by fusesoc
-set fp [open [ls "*.scr"] r]
-set file_data [read $fp]
-close $fp
-set data [split $file_data "\n"]
+set DUT $::env(DUT)
 
-# this TECH_LIB_SEL selects the appropriate technology by defining
+# this PRIM_DEFAULT_IMPL selects the appropriate technology by defining
 # PRIM_DEFAULT_IMPL=prim_pkg::Impl<tech identifier>
-# TECH_LIB_SEL is set inside the library setup script
+# PRIM_DEFAULT_IMPL is set inside the library setup script
 set DEFINE "PRIM_DEFAULT_IMPL=${PRIM_DEFAULT_IMPL} "
```

Comment:
ideally this would be the case, yes. i.e., we would add a define in the `syn` target of the core file.

this is however not done at the moment since we currently do not intend to expose the technology name in the core file that is checked in here.

hence, this define is set in the technology-specific `setup.tcl` further above, and used here to select the correct prim implementation.

Reply:
Aah, I see. thanks for the explanation.